src/daft-catalog/src/identifier.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::Display;
+
+use crate::errors::{Error, Result};
+
+/// An object's namespace (location).
+pub type Namespace = Vec<String>;
+
+/// An object's name.
+pub type Name = String;
```

Comment:
nit: `Name` is pretty ambiguous, could we rename to `Obj
```suggestion
pub type ObjectName = String;
```